msal/src/main/java/com/microsoft/identity/client/internal/controllers/BrokerContentProviderStrategy.java

```diff
@@ -0,0 +1,487 @@
+
+                    @NonNull
+                    @Override
+                    public String getMethodName() {
+                        return methodName;
+                    }
+
+                    @NonNull
+                    @Override
+                    public String getUriPath() {
+                        return ACQUIRE_TOKEN_SILENT_PATH;
+                    }
+
+                    @NonNull
+                    @Override
+                    public AcquireTokenResult getResultFromBundle(Bundle resultBundle) throws BaseException {
+                        return mResultAdapter.getAcquireTokenResultFromResultBundle(resultBundle);
+                    }
+                });
+    }
+
+    @Override
+    List<ICacheRecord> getBrokerAccounts(@NonNull final CommandParameters parameters,
+                                         @Nullable final String negotiatedBrokerProtocolVersion) throws BaseException {
+        final String methodName = "getBrokerAccountsWithContentProvider";
+        return performContentProviderOperation(
+                parameters,
+                new ContentProviderOperation<CommandParameters, List<ICacheRecord>>() {
+            @Nullable
+            @Override
+            public String getRequestString(CommandParameters parameters) {
+                return mRequestAdapter.getRequestStringForGetAccounts(parameters);
+            }
+
+            @Nullable
+            @Override
+            public String[] getOptions() {
+                return new String[]{negotiatedBrokerProtocolVersion};
+            }
+
+            @NonNull
+            @Override
+            public String getMethodName() {
+                return methodName;
+            }
+
+            @NonNull
+            @Override
+            public String getUriPath() {
+                return GET_ACCOUNTS_PATH;
+            }
+
+            @NonNull
+            @Override
+            public List<ICacheRecord> getResultFromBundle(Bundle resultBundle) throws BaseException {
+                return mResultAdapter.getAccountsFromResultBundle(resultBundle);
+            }
+        });
+    }
+
+    @Override
+    void removeBrokerAccount(@NonNull final RemoveAccountCommandParameters parameters,
+                             @Nullable final String negotiatedBrokerProtocolVersion) throws BaseException {
+
+        final String methodName = "removeBrokerAccountWithContentProvider";
+
+        performContentProviderOperation(
+                parameters,
+                new ContentProviderOperation<RemoveAccountCommandParameters, Void>() {
+            @Nullable
+            @Override
+            public String getRequestString(RemoveAccountCommandParameters parameters) {
+                return mRequestAdapter.getRequestStringForRemoveAccount(parameters);
+            }
+
+            @Nullable
+            @Override
+            public String[] getOptions() {
+                return new String[]{negotiatedBrokerProtocolVersion};
```

Comment:
not particularly related to this PR, but if we manage to be able to reuse the request bundle here, can you move negotiatedBrokerProtocolVersion into MsalBrokerRequestAdapter as well? (there will be some change to clean up redundant codes in the other two strategy).

Reply:
resolved via https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1053/commits/9020493baf4c990129af7f4d303efd5a03c85c22